lib/videoroom/room/monitor.ex

```diff
@@ -1,22 +1,32 @@
 defmodule Videoroom.Room.Monitor do
   @moduledoc false
+  use GenServer, restart: :transient
 
   require Membrane.OpenTelemetry
   require Logger
 
-  @spec monitor(pid(), String.t()) :: :ok
-  def monitor(room_pid, room_id) do
+  @spec start_link(list()) :: :ok
+  def start_link([room_pid, room_id]) do
+    GenServer.start_link(__MODULE__, [room_pid, room_id], [])
+  end
+
+  @impl true
+  def init([room_pid, room_id]) do
     ref = Process.monitor(room_pid)
+    {:ok, %{ref: ref, room_pid: room_pid, room_id: room_id}}
```

Comment:
I guess we can hibernate the process as it doesn't do much
```suggestion
    {:ok, %{ref: ref, room_pid: room_pid, room_id: room_id}, :hibernate}
```